pkg/probe/exec/exec.go

```diff
@@ -66,6 +68,16 @@ func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
 			}
 			return probe.Failure, string(data), nil
 		}
+
+		timedoutErr, ok = err.(*TimeoutError)
+		if ok {
+			if utilfeature.DefaultFeatureGate.Enabled(features.ExecProbeTimeout) {
+				return probe.Failure, string(data), nil
+			}
+
+			return probe.Unknown, "", timedoutErr
```

Comment:
this changes old behavior by returning a different error type. Should we keep the original `err` inside the `timedoutErr` to return here?

Just thinking out loud.

Reply:
Updated to return original error, though I think it doesn't matter as long as we return `probe.Unknown` but a non-nil err.